configure.ac

```diff
@@ -1158,6 +1158,47 @@ AC_MSG_RESULT(${enable_linker_map})
 AM_CONDITIONAL([OPENTHREAD_ENABLE_LINKER_MAP], [test "${enable_linker_map}" = "yes"])
 
 #
+# Serial transport
+#
+
+AC_ARG_WITH(serial_transport,
```

Comment:
This configuration seems to be platform specific and only apply to the examples.  What if we used `#define`s in `examples/platforms/nrf52840/platform-config.h` instead?  You can wrap them with `#ifndef` so that it can easily be overridden.

Reply:
We had some discussion internally about this some time ago, and making it global seemed like the only way to disable entire files from the build (usb driver files). But sure, I can hide it within our platform configuration if it's preferred approach.

Reply:
There has been a general desire from the community to avoid using `./configure` for including/excluding source files.  The reason is that it makes life easier for those using other toolchains.  Thoughts?

Reply:
That's a valid argument, I don't have strong opinion against doing it internally, will update the PR to use config `#define`s instead.